bin/spice/cmd/nsql.go

```diff
@@ -0,0 +1,196 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/manifoldco/promptui"
+	"github.com/peterh/liner"
+	"github.com/spf13/cobra"
+	"github.com/spiceai/spiceai/bin/spice/pkg/api"
+	"github.com/spiceai/spiceai/bin/spice/pkg/context"
+	"github.com/spiceai/spiceai/bin/spice/pkg/util"
+)
+
+type NsqlRequest struct {
+	Query string `json:"query"`
+	Model string `json:"model"`
+}
+
+type NsqlResponse struct {
+	Matches    []SearchMatch `json:"matches"`
+	DurationMs uint64        `json:"duration_ms"`
+}
+
+var nsqlCmd = &cobra.Command{
+	Use:   "nsql",
+	Short: "Text-to-SQL REPL",
+	Example: `
+# Start a text-to-sql REPL session with local spiced instance
+spice nsql
+
+# Start a text-to-sql REPL session with spiced instance in spice.ai cloud
+spice nsql --cloud
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		cloud, _ := cmd.Flags().GetBool(cloudKeyFlag)
+		rtcontext := context.NewContext().WithCloud(cloud)
+
+		rtcontext.RequireModelsFlavor(cmd)
+
+		model, err := cmd.Flags().GetString(modelKeyFlag)
+		if err != nil {
+			slog.Error("getting model flag", "error", err)
+			os.Exit(1)
+		}
+		if model == "" {
+			models, err := api.GetData[api.Model](rtcontext, "/v1/models?status=true")
+			if err != nil {
+				slog.Error("listing spiced models", "error", err)
+				os.Exit(1)
+			}
+			if len(models) == 0 {
+				slog.Error("no models found")
+				os.Exit(1)
+			}
+
+			modelsSelection := []string{}
+			selectedModel := models[0].Name
+			if len(models) > 1 {
+				for _, model := range models {
+					modelsSelection = append(modelsSelection, model.Name)
+				}
+
+				prompt := promptui.Select{
+					Label:        "Select model",
+					Items:        modelsSelection,
+					HideSelected: true,
+				}
+
+				_, selectedModel, err = prompt.Run()
+				if err != nil {
+					fmt.Printf("Prompt failed %v\n", err)
+					return
+				}
+			}
+
+			fmt.Println("Using model:\n", selectedModel)
+			model = selectedModel
+		}
+
+		httpEndpoint, err := cmd.Flags().GetString("http-endpoint")
+		if err != nil {
+			slog.Error("getting http-endpoint flag", "error", err)
+			os.Exit(1)
+		}
+		if httpEndpoint != "" {
+			rtcontext.SetHttpEndpoint(httpEndpoint)
+		}
+
+		line := liner.NewLiner()
+		line.SetCtrlCAborts(true)
+		defer line.Close()
+		for {
+			message, err := line.Prompt("nsql> ")
```

Comment:
Before showing the prompt for the first time, we should print some sort of help text describing what the command does, like the `sql` command